Initial pull-request of bodygroup manager

garytate opened the pull request:
/CharEditBodygroup <target>

garytate committed:
Adding bodygroupmanager plugin.

garytate committed:
Added MIT license to bodygroupmanager

garytate committed:
Ported bodygroup to dock usage.

garytate committed:
Accidentally included obfuscated log command

impulsh commented on bodygroupmanager/sv_hooks.lua:
```diff
@@ -0,0 +1,7 @@
+local PLUGIN = PLUGIN
+
+netstream.Hook("ixBodygroupTableSet", function(client, target, bodygroups)
```
You should do some checks here, otherwise anyone can send a net message to change their bodygroups at any time regardless of access

impulsh commented on bodygroupmanager/derma/cl_viewer.lua:
```diff
@@ -0,0 +1,149 @@
+        for _, v in pairs(self.bodygroupIndex) do
+            table.insert(bodygroups, v.index, v.value)
+        end
+        netstream.Start("ixBodygroupTableSet", self.target, bodygroups)
```
You should convert usage of `netstream` to use the built-in `net` library instead. Otherwise, make it clear in the `README` that you require netstream for the plugin to operate

garytate committed:
Added Netstream dependency warning & removed duplicate code.

garytate committed:
Removed netstream dependency

impulsh committed:
Use player instead of character as target

Server-side access check is now consistent with command's access check
Add additional server-side checks
Add missing phrase for command description
Fix log using wrong target
Remove redundant `PLUGIN or {}`

- With permission of author
